# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegVideoFramePool.cpp
# Commit: c3d14f2b234a
# Full Hash: c3d14f2b234a7494edf3231113af52f5fc2d2355
# Author: stransky <stransky@redhat.com>
# Date: 2022-01-25 16:37:35
# Regressor Bug: 1724385
# File Overlap Count: 1
# Description:
#   Bug 1724385 [Linux] Try to create EGLImage over decoded VA-API video frame r=alwu,emilio,media-playback-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D135460
# ==============================================================================

diff -r 5834c469b533 -r c3d14f2b234a dom/media/platforms/ffmpeg/FFmpegVideoFramePool.cpp
--- a/dom/media/platforms/ffmpeg/FFmpegVideoFramePool.cpp	Tue Jan 25 07:03:13 2022 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoFramePool.cpp	Tue Jan 25 07:08:02 2022 +0000
@@ -9,6 +9,10 @@
 #include "mozilla/widget/DMABufLibWrapper.h"
 #include "libavutil/pixfmt.h"
 
+#undef FFMPEG_LOG
+#define FFMPEG_LOG(str, ...) \
+  MOZ_LOG(sPDMLog, mozilla::LogLevel::Debug, (str, ##__VA_ARGS__))
+
 namespace mozilla {
 
 RefPtr<layers::Image> VideoFrameSurfaceDMABuf::GetAsImage() {
@@ -127,6 +131,13 @@
       return nullptr;
     }
     FFMPEG_LOG("Created new VA-API DMABufSurface UID = %d", surface->GetUID());
+    if (!mTextureCreationWorks) {
+      mTextureCreationWorks = Some(surface->VerifyTextureCreation());
+    }
+    if (!*mTextureCreationWorks) {
+      FFMPEG_LOG("  failed to create texture over DMABuf memory!");
+      return nullptr;
+    }
     videoSurface = new VideoFrameSurfaceVAAPI(surface);
     mDMABufSurfaces.AppendElement(videoSurface);
   } else {
@@ -165,6 +176,13 @@
     }
     FFMPEG_LOG("Created new SW DMABufSurface UID = %d", surface->GetUID());
     videoSurface = new VideoFrameSurfaceDMABuf(surface);
+    if (!mTextureCreationWorks) {
+      mTextureCreationWorks = Some(surface->VerifyTextureCreation());
+    }
+    if (!*mTextureCreationWorks) {
+      FFMPEG_LOG("  failed to create texture over DMABuf memory!");
+      return nullptr;
+    }
     mDMABufSurfaces.AppendElement(videoSurface);
     return videoSurface;
   }